array.h

```diff
@@ -1354,6 +1565,7 @@ inline auto make_dense(const shape<>& s) { return s; }
  * The resulting shape may not have `Shape::is_compact` return `true`
  * if the shape has non-compact compile-time constant strides. */
 template <class Shape>
+NDARRAY_DECORATE_IF_CUDA
```

Comment:
Along the lines of resolving functions above... not sure if this is one that should be available in device code. What do you think?

Reply:
Ditto.